tools/setup-dev.sh

```diff
@@ -171,7 +220,36 @@ if ! ${skip_install_deps} ; then
 
 fi
 
-if ! PY_VER_STR=$(python3 --version) ; then
+if $setup_pyenv ; then
```

Comment:
I don't think this is needed. Reduce the use and setup pyenv into a single var. If developer already has pyenv installed then just switch to the pyenv version to 3.8.9 default should work fine.

Reply:
...and if pyenv is not present, but use_pyenv is specified, then it could automatically go run the setup, right?

Reply:
correct, I believe there will be only 3 cases. 
1) distribution already have default python3.8 packages 
2) no python3.8 default packages but developer already using pyenv 
3) no python3.8 default packages need us to install 

So with --use-pyenv flag we either default true to distribution we already know they don't have python3.8 or allow the user to pick the specific version

Reply:
This option is needed, because are following cases:
1) --pyenv --setup-pyenv --python-pyenv 3.8.9 - First time 
2) --pyenv Sequential runs when we do not want even setup pyenv and rebuild python and we want to just rebuild venv
3) --pyenv --python-pyenv 3.8.8 if we do not want to drop existing .pyenv but want to switch and use to python 3.8.8

Reply:
if you look my patch it work the same way. 
--use-pyenv doesn't stop anyone from using even default package exist like TW. 
--use-pyenv doesn't reinstall if existed. 
--use-pyenv switch version base on --python-pyenv if exist. 
So we don't need to split the use and setup. Just use can cover all the cases.